openpathsampling/tests/test_external_snapshots.py

```diff
@@ -0,0 +1,134 @@
+import pytest
+
+import openpathsampling as paths
+import numpy as np
+
+from openpathsampling.engines.external_snapshots.snapshot import (
+    ExternalMDSnapshot
+)
+
+class MockEngine(object):
+    def __init__(self, sequences, sleep_ms=0):
+        self.sequences = sequences
+        self.sleep_ms = sleep_ms
+
+    def read_frame_data(self, filename, position):
+        return self.sequences[filename][position]
+
+
+class ErrorMockEngine(MockEngine):
+    """Mock engine used to create the IndexError in load_details"""
+    def __init__(self, sequences, sleep_ms=0):
+        super(ErrorMockEngine, self).__init__(sequences, sleep_ms)
+        self._sequences = self.sequences
+        self.sequences = {k: [] for k in self.sequences}
+        self.accessed = False
+
+    def read_frame_data(self, filename, position):
+        if self.accessed:
+            self.sequences = self._sequences
+
+        self.accessed = True
+        return super(ErrorMockEngine, self).read_frame_data(filename,
+                                                            position)
+
+
+class TestExternalMDSnapshot(object):
+    def setup(self):
+        self.box = np.array([[1.0, 0.0], [0.0, 1.0]])
+        self.vel = np.array([[1.0, 0.0]])
+
+
+        self.engine = MockEngine(
+            sequences={
+                'foo': [(np.array([[0.0, 0.0]]), self.vel, self.box),
+                        (np.array([[0.1, 0.0]]), self.vel, self.box),
+                        (np.array([[0.2, 0.0]]), self.vel, self.box)]
+            },
+            sleep_ms=0.0
+        )
+        self.snapshots = [
+            ExternalMDSnapshot(file_name='foo',
+                               file_position=i,
+                               engine=self.engine)
+            for i in range(3)
+        ]
+
+    def test_init(self):
+        for (i, snap) in enumerate(self.snapshots):
+            assert snap._reversed is None
+            assert snap.file_name == "foo"
+            assert snap.file_position == i
+            assert snap.engine == self.engine
+            assert snap.velocity_direction == 1
+
+    def test_load_details(self):
+        for (i, snap) in enumerate(self.snapshots):
+            snap.load_details()
+            expected_xyz = np.array([[0.1 * i, 0.0]])
+            expected_vel = np.array([[1.0, 0.0]])
+            np.testing.assert_array_equal(snap._xyz, expected_xyz)
+            np.testing.assert_array_equal(snap._velocities, expected_vel)
+            np.testing.assert_array_equal(snap._box_vectors, self.box)
+
+    def test_load_details_indexerror(self):
+        engine = ErrorMockEngine(self.engine.sequences)
+        engine.accessed
```

Comment:
why is this line here? 
just to test that the attribute is there? 
or is it leftover debug code?

Reply:
Thanks  for catching this: It was there because `ErrorMockEngine` had an `__init_` method instead of an `__init__` method, and it took me an embarrassingly long time to track down the missing underscore. This was part of that process. Removed!

Reply:
ah, classic :laughing: